mmv1/third_party/terraform/services/container/resource_container_cluster.go.tmpl

```diff
@@ -7074,6 +7082,9 @@ func containerClusterAutopilotCustomizeDiff(_ context.Context, d *schema.Resourc
 			return err
 		}
 	}
+	if d.Get("enable_autopilot").(bool) && d.HasChange("dns_config.0.additive_vpc_scope_dns_domain") {
+		return d.ForceNew("dns_config.0.additive_vpc_scope_dns_domain")
+	}
```

Comment:
Can I ask why this check is needed? If I'm reading this right shouldn't the other conditions in the `suppressDiffForAutopilot` sufficiently manage this scenario? I could be wrong or missing something based on your testing.

Reply:
Sorry if I overcomplicated it, it was my first Terraform code and I'm not on friendly terms with the documentation. To be clear what I'm trying to accomplish between the diff suppress and here is that 1. the only thing you are allowed to customize in Autopilot clusters is the additive VPC domain 2. after creation of an Autopilot cluster no modifications to DNS config are permitted by container API, any update involving these will fail.

So with that in mind, is what you're saying that I could just use ForceNew in the suppress diff function?

Reply:
Ah! Your context makes sense. Thanks for that. 

I believe you designed it correctly then. I don't believe it's possible to forceNew from a suppress diff function.